VMBackup/main/Utils/EventLoggerUtil.py

```diff
@@ -0,0 +1,256 @@
+import os
+import threading
+import json
+import sys
+import datetime
+import time
+if sys.version_info[0] == 2:
+    import Queue as queue
+else:
+    import queue
+import shutil
+from Utils.LogHelper import FileHelpers,LoggingConstants
+from Utils.StringHelper import StringHelper
+from Utils.Event import Event
+
+class EventLogger:
+    _instance = None
+    _lock = threading.Lock()
+    
+    
+    def __init__(self, log, event_directory, severity_level):
+        self.temporary_directory = os.path.join(event_directory, 'Temp')
+        self.space_available_in_event_directory = 0
+        self.event_processing_interval = 0
+        self.logger = log
+        self.disposed = False
+        self.event_processing_task = None  
+        self.current_message_len = 0
+        self.event_logging_enabled = False
+        self.event_logging_error_count = 0
+        self.events_folder = event_directory
+        self.event_logging_enabled = bool(self.events_folder)
+        self.filehelper = FileHelpers()
+
+        if self.event_logging_enabled:
+            self.extension_version = os.path.basename(os.getcwd())
+            self.operation_id = ''
```

Comment:
To keep consistent with Windows, lets make it an emtpy guid which will have all Zeros.